iceoryx_hoofs/include/iceoryx_hoofs/internal/containers/uninitialized_array.inl

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_INL
+#define IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_INL
+
+#include "iceoryx_hoofs/internal/containers/uninitialized_array.hpp"
+
+namespace iox
+{
+namespace containers
+{
+template <typename ElementType, uint64_t Capacity, template <typename, uint64_t> class Buffer>
+inline constexpr ElementType&
+UninitializedArray<ElementType, Capacity, Buffer>::operator[](const uint64_t index) noexcept
+{
+    // AXIVION Next Construct AutosarC++19_03-A5.2.3 : const_cast to avoid code duplication
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
+    return const_cast<ElementType&>(const_cast<const UninitializedArray*>(this)->operator[](index));
```

Comment:
With C++17 we could use `std::as_const` as well. See https://stackoverflow.com/questions/123758/how-do-i-remove-code-duplication-between-similar-const-and-non-const-member-func

A little less ugly but functionally the same.

Resolve me.